sqlalchemy_redshift/commands.py

```diff
@@ -18,10 +18,21 @@
 ACCESS_KEY_ID_RE = re.compile('[A-Z0-9]{20}')
 SECRET_ACCESS_KEY_RE = re.compile('[A-Za-z0-9/+=]{40}')
 TOKEN_RE = re.compile('[A-Za-z0-9/+=]+')
+IAM_ROLE_RE = re.compile('arn:aws:iam::[0-9]+:role/.+')
 
 
-def _process_aws_credentials(access_key_id, secret_access_key,
-                             session_token=None):
+def _process_aws_credentials(access_key_id=None, secret_access_key=None,
+                             session_token=None, iam_role=None):
+
+    if iam_role is not None:
+        if not IAM_ROLE_RE.match(iam_role):
+            raise ValueError(
+                'invalid IAM role ARN; does not match {pattern}'.format(
+                    pattern=IAM_ROLE_RE.pattern,
+                )
+            )
+        else:
+            return 'aws_iam_role=' + iam_role
```

Comment:
I'd like to stick with the code as is for now, maybe add a TypeError

I've added it to my personal v1-breaking-change-wishlist